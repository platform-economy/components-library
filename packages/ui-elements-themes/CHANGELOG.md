# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
