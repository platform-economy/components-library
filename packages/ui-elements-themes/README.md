# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
