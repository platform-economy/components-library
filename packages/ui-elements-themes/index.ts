export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
