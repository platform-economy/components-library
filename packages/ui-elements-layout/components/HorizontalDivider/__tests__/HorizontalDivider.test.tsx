import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { withTestTheme } from '@relayr/ui-elements-themes';
/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';

// Component
import { HorizontalDivider } from '../index';

describe('ui-elements-layout', () => {
  describe('Components', () => {
    describe('HorizontalDivider', () => {
      it('Should render with default state', () => {
        const tree = TestRenderer
          .create(withTestTheme(<HorizontalDivider />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Should render with starting and ending rules', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <HorizontalDivider startingRule={true} endingRule={true}>
              <span>left</span>
              <span>center</span>
              <span>right</span>
            </HorizontalDivider>,
          ))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Should render only one rule', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <HorizontalDivider>
              <span>left</span>
              <span>right</span>
            </HorizontalDivider>,
          ))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
