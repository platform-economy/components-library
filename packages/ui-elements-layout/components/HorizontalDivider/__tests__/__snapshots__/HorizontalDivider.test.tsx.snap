// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-layout Components HorizontalDivider Should render only one rule 1`] = `
.c1 {
  border: none;
  border-bottom: 1px solid;
  color: #divider_inverted;
  width: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > hr {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 0;
  margin: 0 20px;
}

.c0 > *:first-child {
  margin-left: 0;
}

.c0 > *:last-child {
  margin-right: 0;
}

<div
  className="c0"
>
  <span>
    left
  </span>
  <hr
    className="c1"
  />
  <span>
    right
  </span>
</div>
`;

exports[`ui-elements-layout Components HorizontalDivider Should render with default state 1`] = `
.c1 {
  border: none;
  border-bottom: 1px solid;
  color: #divider_inverted;
  width: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > hr {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 0;
  margin: 0 20px;
}

.c0 > *:first-child {
  margin-left: 0;
}

.c0 > *:last-child {
  margin-right: 0;
}

<div
  className="c0"
>
  <hr
    className="c1"
  />
</div>
`;

exports[`ui-elements-layout Components HorizontalDivider Should render with starting and ending rules 1`] = `
.c1 {
  border: none;
  border-bottom: 1px solid;
  color: #divider_inverted;
  width: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > hr {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 0;
  margin: 0 20px;
}

.c0 > *:first-child {
  margin-left: 0;
}

.c0 > *:last-child {
  margin-right: 0;
}

<div
  className="c0"
>
  <hr
    className="c1"
  />
  <span>
    left
  </span>
  <hr
    className="c1"
  />
  <span>
    center
  </span>
  <hr
    className="c1"
  />
  <span>
    right
  </span>
  <hr
    className="c1"
  />
</div>
`;
